dspace-api/src/main/java/org/dspace/statistics/AnonymizeStatistics.java

```diff
@@ -0,0 +1,274 @@
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.log4j.Logger;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.dspace.core.Context;
+import org.dspace.services.ConfigurationService;
+import org.dspace.services.factory.DSpaceServicesFactory;
+import org.dspace.statistics.factory.StatisticsServiceFactory;
+import org.dspace.statistics.service.SolrLoggerService;
+
+public class AnonymizeStatistics {
+
+    private static Logger log = getLogger(AnonymizeStatistics.class);
+    private static Context context = new Context();
+    private static String action = "anonymise_statistics";
+
+    private static final String HELP_OPTION = "h";
+    private static final String SLEEP_OPTION = "s";
+    private static final String BATCH_SIZE_OPTION = "b";
+    private static final String THREADS_OPTION = "t";
+
+    private static int sleep;
+
+    private static SolrLoggerService solrLoggerService =
+            StatisticsServiceFactory.getInstance().getSolrLoggerService();
+    private static ConfigurationService configurationService =
+            DSpaceServicesFactory.getInstance().getConfigurationService();
+
+    private static int batchSize = 100;
+    private static int threads = 2;
+
+    private static final Object ANONYMISED =
+            configurationService.getProperty("anonymise_statistics.dns_mask", "anonymised");
+
+    private static final String TIME_LIMIT;
+
+    static {
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(DAY_OF_YEAR, -configurationService.getIntProperty("anonymise_statistics.time_limit", 90));
+        TIME_LIMIT = format(calendar, DATE_FORMAT_8601);
+    }
+
+    private AnonymizeStatistics() {
+
+    }
+
+
+    public static void main(String... args) throws ParseException {
+
+        parseCommandLineOptions(createCommandLineOptions(), args);
+        anonymiseStatistics();
+    }
+
+    private static Options createCommandLineOptions() {
+
+        Options options = new Options();
+        options.addOption(
+                builder(HELP_OPTION)
+                        .longOpt("help")
+                        .desc("Print the usage of the script")
+                        .hasArg(false)
+                        .build()
+        );
+
+        options.addOption(
+                builder(SLEEP_OPTION)
+                        .longOpt("sleep")
+                        .desc("Sleep a certain time between each solr request")
```

Comment:
suggestion: rephrase to "Sleep a certain time **given in milliseconds** between each solr request"

Reply:
+1 for this suggestion

Reply:
Good suggestion, I've changed it.